### PR TITLE
[MOD-2611] Expose information to `INFO MODULES` command

### PR DESCRIPTION
**Pull request by Avital-Fine**

### CODE REVIEW

File: src/config.c [hunk 1]
```diff
@@ -844,6 +844,33 @@ int RSConfig_SetOption(RSConfig *config, RSConfigOptions *options, const char *n
   return rc;
 }
 
+void RSConfig_AddToInfo(RedisModuleInfoCtx *ctx) {
+  RedisModule_InfoAddSection(ctx, "run_time_configs");
```
Comment (ashtul) [inline src/config.c:848]: Some of these aren't strictly run time configs.
Maybe a better name would `module configurations`?
@K-Jo WDYT?

File: src/fields_global_stats.c [hunk 2]
```diff
@@ -0,0 +1,92 @@
...
+    else if (fs->types & INDEXFLD_T_GEO) RSGlobalConfig.fieldsStats.numGeoFieldsSortable += toAdd;
+    else if (fs->types & INDEXFLD_T_TAG) RSGlobalConfig.fieldsStats.numTagFieldsSortable += toAdd;
+  }
+  if (fs->options & FieldSpec_NotIndexable) {
+    if (fs->types & INDEXFLD_T_FULLTEXT) RSGlobalConfig.fieldsStats.numTextFieldsNoIndex += toAdd;
+    else if (fs->types & INDEXFLD_T_NUMERIC) RSGlobalConfig.fieldsStats.numNumericFieldsNoIndex += toAdd;
+    else if (fs->types & INDEXFLD_T_GEO) RSGlobalConfig.fieldsStats.numGeoFieldsNoIndex += toAdd;
+    else if (fs->types & INDEXFLD_T_TAG) RSGlobalConfig.fieldsStats.numTagFieldsNoIndex += toAdd;
+  }
+}
```
Comment (ashtul) [inline src/fields_global_stats.c:36]: Lets use a switch case.
I think we remove the NoIndex fields. @K-jo, WDYT?

File: src/module-init/module-init.c [hunk 3]
```diff
@@ -194,10 +222,18 @@ int RediSearch_Init(RedisModuleCtx *ctx, int mode) {
...
+    return REDISMODULE_ERR;
+  }
+
   Initialize_KeyspaceNotifications(ctx);
   Initialize_CommandFilter(ctx);
   GetJSONAPIs(ctx, 1);
   Initialize_RdbNotifications(ctx);
+  RSGlobalConfig.fieldsStats.numTextFields = 0;
+  RSGlobalConfig.fieldsStats.numTextFieldsNoIndex = 0;
+  RSGlobalConfig.fieldsStats.numTextFieldsSortable = 0;
```
Comment (ashtul) [inline src/module-init/module-init.c:236]: Why only these fields?

File: src/module.c [hunk 4]
```diff
@@ -114,7 +114,7 @@ int SpellCheckCommand(RedisModuleCtx *ctx, RedisModuleString **argv, int argc) {
     return RedisModule_WrongArity(ctx);
   }
 
-  
+
```
Comment (ashtul) [inline src/module.c:117]: remove line

File: src/redis_index.c [hunk 5]
```diff
@@ -478,16 +478,15 @@ int Redis_DeleteKeyC(RedisModuleCtx *ctx, char *cstr) {
...
   SchemaPrefixes_RemoveSpec(spec);
-
   if (deleteDocuments || !!(spec->flags & Index_Temporary)) {
     DocTable *dt = &spec->docs;
     DOCTABLE_FOREACH(dt, Redis_DeleteKeyC(ctx->redisCtx, dmd->keyPtr));
   }
-
+  // for (int i = 0; i < spec->numFields; i++) {
+  //   FieldSpec_UpdateGlobalStat(spec->fields + i, -1);
+  // }
```
Comment (ashtul) [inline src/redis_index.c:489]: remove

File: src/spec.h [hunk 6]
```diff
@@ -541,6 +545,8 @@ typedef struct IndexesScanner {
   bool cancelled;
 } IndexesScanner;
 
+double IndexesScanner_IndexedPrecent(IndexesScanner *scanner, IndexSpec *sp);
```
Comment (ashtul) [inline src/spec.h:548]: Precent -> Percent

**Review by ashtul — COMMENTED** [review]
The PR looks good.
Some minor comments.
Let's also add a test for reloading from RDB using `retry_with_rdb_reload`.

File: src/fields_global_stats.c (hunk 2 above)
Comment (Avital-Fine) [reply src/fields_global_stats.c:36]: @ashtul Is it possible to do switch case to & operator?

Comment (ashtul) [reply src/fields_global_stats.c:36]: You are right.
We can have multi typed fields

**Review by ashtul — DISMISSED** [review]
LGTM

**Review by ashtul — APPROVED** [review]
Looks great

**Pull request merged**